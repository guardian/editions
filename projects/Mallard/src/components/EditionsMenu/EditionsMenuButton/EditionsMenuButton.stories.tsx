import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { EditionsMenuButton } from './EditionsMenuButton';

storiesOf('EditionsMenu', module)
	.addDecorator(withKnobs)
	.add('EditionsMenuButton - default', () => (
		<EditionsMenuButton onPress={() => {}} />
	))
	.add('EditionsMenuButton - selected', () => (
		<EditionsMenuButton onPress={() => {}} selected />
	));
