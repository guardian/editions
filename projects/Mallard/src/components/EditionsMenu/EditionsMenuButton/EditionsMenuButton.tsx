import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { Editions } from 'src/components/icons/Editions';
import { LeftChevron } from 'src/components/icons/LeftChevron';
import { color } from 'src/theme/color';
import { metrics } from 'src/theme/spacing';
import { getFont } from 'src/theme/typography';

const styles = (selected: boolean) =>
	StyleSheet.create({
		button: {
			backgroundColor: selected
				? color.palette.sport.pastel
				: 'transparent',
			borderRadius: 24,
			justifyContent: 'center',
			height: 42,
			width: 42,
		},
		iconContainer: {
			flex: 1,
			justifyContent: 'space-around',
			width: 55,
			marginRight: metrics.horizontal,
		},
		label: {
			color: 'white',
			...getFont('sans', 0.5),
			fontSize: 12,
			lineHeight: 14.75,
			top: 8,
		},
	});

const EditionsMenuButton = ({
	disabled = false,
	onPress,
	selected = false,
}: {
	disabled?: boolean;
	onPress: () => void;
	selected?: boolean;
}) => (
	<TouchableOpacity
		disabled={disabled}
		accessibilityRole="button"
		accessibilityLabel="Regions and specials editions menu"
		onPress={onPress}
		style={styles(selected).button}
	>
		{selected ? (
			<LeftChevron />
		) : (
			<View style={styles(selected).iconContainer}>
				<Editions />

				<Text style={styles(selected).label}>Editions</Text>
			</View>
		)}
	</TouchableOpacity>
);

export { EditionsMenuButton };
