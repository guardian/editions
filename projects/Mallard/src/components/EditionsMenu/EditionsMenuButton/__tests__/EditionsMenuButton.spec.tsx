import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { EditionsMenuButton } from '../EditionsMenuButton'

describe('EditionsMenuButton', () => {
    it('should display a default EditionsMenuButton with correct styling', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <EditionsMenuButton onPress={() => {}} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a selected EditionsMenuButton with correct styling', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <EditionsMenuButton selected onPress={() => {}} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
