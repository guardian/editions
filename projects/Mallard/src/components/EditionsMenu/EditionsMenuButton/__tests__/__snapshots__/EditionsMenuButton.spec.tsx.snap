// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditionsMenuButton should display a default EditionsMenuButton with correct styling 1`] = `
<View
  accessibilityLabel="Regions and specials editions menu"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 24,
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "width": 42,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
        "marginRight": 14,
        "width": 55,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="42"
      bbWidth="42"
      fill="none"
      focusable={false}
      height={42}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 42,
            "width": 42,
          },
        ]
      }
      width={42}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            Object {
              "payload": 4278190080,
              "type": 0,
            }
          }
          height="41"
          propList={
            Array [
              "stroke",
            ]
          }
          rx="20.5"
          stroke={
            Object {
              "payload": 4294967295,
              "type": 0,
            }
          }
          width="41"
          x="0.5"
          y="0.5"
        />
        <RNSVGCircle
          cx="29"
          cy="14"
          fill={
            Object {
              "payload": 4294949576,
              "type": 0,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          r="7"
        />
        <RNSVGCircle
          cx="14"
          cy="14"
          fill={
            Object {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          r="7"
        />
        <RNSVGCircle
          cx="29"
          cy="29"
          fill={
            Object {
              "payload": 4287683839,
              "type": 0,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          r="7"
        />
        <RNSVGCircle
          cx="14"
          cy="29"
          fill={
            Object {
              "payload": 4294934287,
              "type": 0,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          r="7"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "GuardianTextSans-Regular",
          "fontSize": 12,
          "lineHeight": 14.75,
          "top": 8,
        }
      }
    >
      Editions
    </Text>
  </View>
</View>
`;

exports[`EditionsMenuButton should display a selected EditionsMenuButton with correct styling 1`] = `
<View
  accessibilityLabel="Regions and specials editions menu"
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#90dcff",
      "borderRadius": 24,
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "width": 42,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#333333",
        "fontFamily": "GuardianIcons-Regular",
        "fontSize": 20,
        "lineHeight": 20,
      }
    }
  >
    î€Š
  </Text>
</View>
`;
